### PR TITLE
734 access internal jacobians

### LINKED ISSUE #734: Access internal jacobians
There are a number of minimizers that can use an internal jacobian that can't be exported.  Some of these are hard coded, e.g., `lm-scipy-no-jac`, `levmar-no-jac`, whereas we currently have no reliable way to access the others (e.g., Mantid, RALFit,....)

We should add an option to the Jacobian options called 'solver-default', which:
- uses the derivatives bundled with the solver if possible, otherwise
- uses our default

Once this is added, we can remove the minimizers 'lm-scipy-no-jac', 'levmar-no-jac'.

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Fixes #734 by adding a new Jacobian option, ``default``, which uses the Jacobian supplied by the solver in the case that no Jacobian is supplied.  This replaces the `<solver>-no-jac` functionality that was present to access this before, and those solvers have been removed.

I've also updated some of the tests to make them easier to update/troubleshoot.

#### Testing Instructions

1. Run tests with `jac_method = default`, and check results are as expected
2. Check results are the same as for `scipy_no_jac`
3. Read through the docs and check they are clear

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

**Comment by tyronerees** [comment]
Here are the results for NIST (average difficulty) before removing `*-no-jac`:
image

In summary, `scipy-no-jac` and the default Jacobian give the same behaviour for accuracy (as expected).  `levmar-no-jac` and `levmar` with the default Jacobian are different.  `levmar` with the default Jacobian seems more robust, which suggest the issues with levmar are with the Jacobian we're passing in.

[NIST_average_difficulty.zip](https://github.com/fitbenchmarking/fitbenchmarking/files/6348830/NIST_average_difficulty.zip)

File: docs/source/users/options/jacobian_option.rst [hunk 1]
```diff
@@ -42,6 +42,25 @@ Default is ``2-point``
...
 
+.. _defaultjacobian:
+
+Solver Default Jacobian (:code:`default`)
+--------------------------------------------
+
+This uses the approximation of the Jacobian that is used by default in the minimizer,
+and will vary between solvers.  If the minimizer requires the user to pass a Jacobian,
+a warning will be printed to the screen and the :ref:`scipy-jac` 2-point
+approximation will be used.  The only options is:
```
Comment (jess-farmer) [inline docs/source/users/options/jacobian_option.rst:53]:
```suggestion
approximation will be used.  The only option is:
```

File: examples/benchmark_problems/SAS_modelling/1D/prob_def_1.txt [hunk 2]
```diff
@@ -4,4 +4,4 @@ name = 'Problem Def 1'
 description = 'An example data set for SasView 1D data'
 input_file = 'cyl_400_20.txt'
 function ='name=cylinder,radius=35.0,length=350.0,background=0.0,scale=1.0,sld=4.0,sld_solvent=1.0'
-parameter_ranges = {'radius': (1,50), 'length': (1,500)}
\ No newline at end of file
+parameter_ranges = {'radius': (30,40), 'length': (1,500)}
```
Comment (jess-farmer) [inline examples/benchmark_problems/SAS_modelling/1D/prob_def_1.txt:7]: Assuming this change has been accidentally saved

```suggestion
parameter_ranges = {'radius': (1,50), 'length': (1,500)}
```

File: fitbenchmarking/utils/options.py [hunk 3]
```diff
@@ -35,18 +35,18 @@ class Options(object):
...
          'software': ['bumps', 'dfo', 'gsl', 'levmar', 'mantid', 'minuit',
                       'ralfit', 'scipy', 'scipy_ls'],
-         'jac_method': ['scipy', 'analytic', 'numdifftools'],
+         'jac_method': ['analytic', 'scipy', 'default', 'numdifftools'],
          'cost_func_type': ['nlls', 'weighted_nlls', 'hellinger_nlls',
                             'poisson']}
     VALID_JACOBIAN = \
         {'scipy': ['2-point', '3-point', 'cs'],
          'analytic': ['cutest'],
+         'default': ['default'],
```
Comment (jess-farmer) [inline fitbenchmarking/utils/options.py:49]: This isn't consistent with the docs, where 'default' is replaced with 'solver'

Screenshot 2021-05-10 133239

**Review by jess-farmer — CHANGES_REQUESTED** [review]
Looks good and works as expected, only a few very minor comments

File: examples/benchmark_problems/SAS_modelling/1D/prob_def_1.txt (hunk 2 above)
Comment (tyronerees) [reply examples/benchmark_problems/SAS_modelling/1D/prob_def_1.txt]: Good spot -- thanks!

**Review by jess-farmer — APPROVED** [review]

**Pull request merged**